x-pack/plugins/security_solution/server/lib/timeline/routes/notes/get_notes.ts

```diff
@@ -43,78 +45,90 @@ export const getNotesRoute = (router: SecuritySolutionPluginRouter) => {
-                page: 1,
-                perPage: maxUnassociatedNotes,
-              };
-              const res = await getAllSavedNote(frameworkRequest, options);
-              return response.ok({ body: res ?? {} });
             }
-          } else if (savedObjectIds != null) {
+
+            // searching for all the notes associated with a specific document id
+            const options: SavedObjectsFindOptions = {
+              type: noteSavedObjectType,
+              filter: nodeBuilder.is(`${noteSavedObjectType}.attributes.eventId`, documentIds),
+              page: 1,
+              perPage: maxUnassociatedNotes,
+            };
+            const res = await getAllSavedNote(frameworkRequest, options);
+            return response.ok({ body: res ?? {} });
+          }
+
+          // if savedObjectIds is provided, we will search for all the notes associated with the savedObjectIds
+          const savedObjectIds = queryParams.savedObjectIds ?? null;
+          if (savedObjectIds != null) {
+            // search for multiple saved object ids
             if (Array.isArray(savedObjectIds)) {
-              const soIdSearchString = savedObjectIds?.join(' | ');
-              const options = {
+              const options: SavedObjectsFindOptions = {
                 type: noteSavedObjectType,
-                hasReference: {
+                hasReference: savedObjectIds.map((savedObjectId: string) => ({
                   type: timelineSavedObjectType,
-                  id: soIdSearchString,
-                },
+                  id: savedObjectId,
+                })),
                 page: 1,
                 perPage: maxUnassociatedNotes,
               };
               const res = await getAllSavedNote(frameworkRequest, options);
               const body: GetNotesResponse = res ?? {};
               return response.ok({ body });
-            } else {
-              const options = {
-                type: noteSavedObjectType,
-                hasReference: {
-                  type: timelineSavedObjectType,
-                  id: savedObjectIds,
-                },
-                perPage: maxUnassociatedNotes,
-              };
-              const res = await getAllSavedNote(frameworkRequest, options);
-              const body: GetNotesResponse = res ?? {};
-              return response.ok({ body });
             }
-          } else {
-            const perPage = queryParams?.perPage ? parseInt(queryParams.perPage, 10) : 10;
-            const page = queryParams?.page ? parseInt(queryParams.page, 10) : 1;
-            const search = queryParams?.search ?? undefined;
-            const sortField = queryParams?.sortField ?? undefined;
-            const sortOrder = (queryParams?.sortOrder as SortOrder) ?? undefined;
-            const filter = queryParams?.filter;
-            const options = {
+
+            // searching for all the notes associated with a specific for saved object id
```

Comment:
```suggestion
            // searching for all the notes associated with a specific saved object id
```

Reply:
fixed in [this commit](https://github.com/elastic/kibana/pull/196225/commits/fcd19df73e926148dce5a070d467bd8378184a63)